### PR TITLE
Modernize csproj

### PR DESCRIPTION
**Pull request by js6pak**
- Migrate to the new .csproj format
- Fix tests on non-windows platforms

### CODE REVIEW

**Comment by willardf** [comment]
The tests no longer pass on Windows now. All DtlsConnectionTests fail with

```
Test method Hazel.UnitTests.Dtls.ConnectionTests.ClientDisconnectTest threw exception: 
System.ArgumentException: Certificate must be signed by an RSA key
Parameter name: certificate
```

File: Hazel/ConnectionListener.cs [hunk 1]
```diff
@@ -79,10 +79,6 @@ protected void InvokeNewConnection(MessageReader msg, Connection connection)
                 {
                 }
             }
-            else
```
Comment (willardf) [inline Hazel/ConnectionListener.cs:82]: This change seems unrelated and wrong.

File: Hazel.UnitTests/Hazel.UnitTests.csproj [hunk 2]
```diff
@@ -1,122 +1,16 @@
...
-  </Target>
-  <Target Name="AfterBuild">
-  </Target>
-  -->
-</Project>
\ No newline at end of file
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net472</TargetFramework>
```
Comment (willardf) [inline Hazel.UnitTests/Hazel.UnitTests.csproj:4]: I have concerns about moving to 4.7.2 I don't think it'll be compatible with Unity, but I'll have to double check.

File: Hazel/ConnectionListener.cs (hunk 1 above)
Comment (js6pak) [reply Hazel/ConnectionListener.cs:82]: While it is a bit out of scope, it's impossible to pass the tests with DEBUG mode without it. [InvokeNewConnection](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/ConnectionListener.cs#L68) [recycles](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/ConnectionListener.cs#L84) the hello reader, it's only ever called by [ReadCallback](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/Udp/UdpConnectionListener.cs#L225) which in turn always calls [HandleReceive](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/Udp/UdpConnectionListener.cs#L229) which always [disposes](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/Udp/UdpConnection.cs#L150) second time causing an [exception](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/ObjectPool.cs#L102-L104) every time a connection connects without anything subscribed to `NewConnection` event.

File: Hazel.UnitTests/Hazel.UnitTests.csproj (hunk 2 above)
Comment (js6pak) [reply Hazel.UnitTests/Hazel.UnitTests.csproj:4]: You've referenced the `UnitTests` project which always runs on your host's .NET runtime so it's not really compatible with Unity anyway. But afaik Unity supports net472 for a long time now

File: Hazel/ConnectionListener.cs (hunk 1 above)
Comment (willardf) [reply Hazel/ConnectionListener.cs:82]: Alright yeah, I'm on board. This is just something I missed from my breaking change. Interestingly it doesn't break any test or get captured in any way, but it is a good change.

**Review by willardf — APPROVED** [review]
